### PR TITLE
Applications

### PR DESCRIPTION
**Pull request by sarynm12**
- Applications section finished 
- All tests passing
- 100% coverage

### CODE REVIEW

File: app/views/applications/apply.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    <label>Phone Number:</label><br>
+    <%= text_field_tag :phone_number%><br>
+
+    <label>Why would you make a good home for this/these pet(s)?</label><br>
+    <%= text_field_tag :description%><br><br>
+
+    <%= submit_tag "Submit Application" %><br><br>
+  <% end %>
+</div>
```
Comment (lkriffell) [inline app/views/applications/apply.html.erb:35]: Nice. I see a user can apply to adopt a pet or multiple pets

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -117,3 +115,34 @@
...
+  text-align: left;
+}
+
+.dropdown:hover .dropdown-content {
+  display: block;
+}
+
+.links {
+
+}
```
Comment (fieldstyler) [inline app/assets/stylesheets/application.css:148]: Good work on the dropdowns

File: app/controllers/applications_controller.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      checked_pets.each do |checked_pet|
+        ApplicationsPets.create!({applications_id: new_app.id, pets_id: checked_pet.id})
+      end
+      flash[:notice] = "Your application for #{@pets_applied_for} has been submitted!"
+      redirect_to '/favorites'
+    else
+      flash[:alert] = "You must select a pet to apply for."
+      redirect_to '/favorites/apply'
+    end
+  end
```
Comment (fieldstyler) [inline app/controllers/applications_controller.rb:50]: Let's refactor this for sure.

File: app/controllers/applications_controller.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  def collect_checked_pets
+    ids = Pet.all.select(:id)
+    checked_pets = []
+    ids.each do |id|
+      if params.include?(id.id.to_s)
+        checked_pets << id
+      end
+    end
+    checked_pets
+  end
```
Comment (fieldstyler) [inline app/controllers/applications_controller.rb:61]: Nice helper method

File: app/views/applications/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      <ul>
+        <% if pet.adoption_status == "Adoptable" %>
+        <li id=<%="checked_field-#{pet.id}"%>><%= link_to "#{pet.name}", "/pets/#{pet.id}" %> - <%= check_box_tag "#{pet.id}", "#{pet.name}"%></li>
+        <% else %>
+          <li><%= link_to "#{pet.name}", "/pets/#{pet.id}" %> - This pet is pending adoption. - <%= link_to "Revoke Application", "/applications/#{@app.id}/#{pet.id}", method: :patch %></li>
+        <% end %>
+      </ul>
+    <% end %>
+    <li><%= submit_tag "Approve Application" %></li>
+  <% end %>
```
Comment (lkriffell) [inline app/views/applications/show.html.erb:27]: Awesome. I see we can approve an app for multiple pets and then revoke an application for a specific pet

File: app/controllers/applications_controller.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+  def approve
+    app = Application.find(params[:id])
+    checked_pets = collect_checked_pets
+    checked_pets.each do |checked_pet|
+      pet = Pet.find(checked_pet.id)
+      pet.update!(adoption_status: "Pending")
+      pet.save!
+      #require "pry"; binding.pry
```
Comment (fieldstyler) [inline app/controllers/applications_controller.rb:70]: remove comments

File: app/controllers/applications_pets_controller.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class ApplicationsPetsController < ApplicationRecord
+
+  def abcd
```
Comment (fieldstyler) [inline app/controllers/applications_pets_controller.rb:3]: need to change method name

File: app/controllers/pets_controller.rb [hunk 8]
```diff
@@ -58,6 +63,12 @@ def add_favorite
 
   def show_favorite
     @favorites = Pet.favorited_pets
+    @app_pets = ApplicationsPets.all
+    a = []
+    @app_pets.each do |pet|
+      a << pet.pets_id
```
Comment (fieldstyler) [inline app/controllers/pets_controller.rb:69]: change naming

File: app/views/pets/show.html.erb [hunk 9]
```diff
@@ -1,20 +1,44 @@
...
+      <% end %>
+      <% @apps_pets.each do |app_pet| %>
+      <% app = Application.get_app_by_id(app_pet.applications_id) %>
+      <ul>
+          <li><%= link_to "#{app.name}", "/applications/#{app.id}" %></li>
+      </ul>
+      <% end %>
+    </div>
+  </div>
+
```
Comment (lkriffell) [inline app/views/pets/show.html.erb:20]: Nice dropdown menu to display a list of apps for a pet

File: db/schema.rb [hunk 10]
```diff
@@ -10,11 +10,38 @@
...
+    t.string "description"
+  end
+
+  create_table "applications_pets", force: :cascade do |t|
+    t.bigint "applications_id"
+    t.bigint "pets_id"
+    t.index ["applications_id"], name: "index_applications_pets_on_applications_id"
+    t.index ["pets_id"], name: "index_applications_pets_on_pets_id"
+  end
+
```
Comment (lkriffell) [inline db/schema.rb:44]: Nice many to many joins table

File: app/views/applications/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+</ul>
+
+  <h3>Pets Applied For:</h3>
+  <p>Check each pet you'd like to approve. Pending pets cannot be approved.</p>
+  <%= form_tag ("/applications/#{@app.id}") do %>
+    <% @apps_pets.each do |app_pet| %>
+    <% pet = Pet.get_pet_by_id(app_pet.pets_id) %>
+      <ul>
+        <% if pet.adoption_status == "Adoptable" %>
+        <li id=<%="checked_field-#{pet.id}"%>><%= link_to "#{pet.name}", "/pets/#{pet.id}" %> - <%= check_box_tag "#{pet.id}", "#{pet.name}"%></li>
```
Comment (fieldstyler) [inline app/views/applications/show.html.erb:20]: love how dynamic this is

File: spec/features/applications/apply_spec.rb [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+    fill_in :phone_number, with: "234-735-4743"
+    fill_in :description, with: "I'm the best."
+
+    click_on 'Submit Application'
+
+    expect(current_path).to eq("/favorites")
+    expect(page).to have_content("Your application for #{@pet1.name} and #{@pet2.name} has been submitted!")
+  end
+
+end
```
Comment (lkriffell) [inline spec/features/applications/apply_spec.rb:179]: Nice test. Pretty robust

**Comment by fieldstyler** [comment]
Reviewed by: Tyler and Logan

**Comment by lkriffell** [comment]
Reviewed by: Logan and Tyler

**Pull request merged**